db/db.go

```diff
@@ -126,3 +152,53 @@ func handleRTDBError(resp *internal.Response) error {
 
 	return err
 }
+
+// parseURLConfig returns the dbURLConfig for the database
+// dbURL may be either:
+//   - a production url (https://foo-bar.firebaseio.com/)
+//   - an emulator URL (localhost:9000/?ns=foo-bar)
+//
+// The following rules will apply for determining the output:
+//   - If the url has no scheme it will be assumed to be an emulator url and be used.
+//   - else If the FIREBASE_DATABASE_EMULATOR_HOST environment variable is set it will be used.
+//   - else the url will be assumed to be a production url and be used.
+func parseURLConfig(dbURL string) (*dbURLConfig, bool, error) {
+	parsedURL, err := url.ParseRequestURI(dbURL)
+	if err != nil {
+		return nil, false, fmt.Errorf("%s: %w", dbURL, ErrInvalidURL)
+	}
+	if parsedURL.Scheme != "https" {
+		cfg, err := parseEmulatorHost(dbURL, parsedURL)
+		return cfg, true, err
+	}
+	return &dbURLConfig{
+		BaseURL:   dbURL,
+		Namespace: "",
+	}, false, nil
+}
+
+func parseEmulatorHost(rawEmulatorHostURL string, parsedEmulatorHost *url.URL) (*dbURLConfig, error) {
+	if strings.Contains(rawEmulatorHostURL, "//") {
+		return nil, fmt.Errorf(`invalid %s: "%s". It must follow format "host:port": %w`, emulatorDatabaseEnvVar, rawEmulatorHostURL, ErrInvalidURL)
+	}
+
+	baseURL := strings.Replace(rawEmulatorHostURL, fmt.Sprintf("?%s", parsedEmulatorHost.RawQuery), "", -1)
+	if parsedEmulatorHost.Scheme != "http" {
+		baseURL = fmt.Sprintf("http://%s", baseURL)
+	}
+
+	namespace := parsedEmulatorHost.Query().Get(emulatorNamespaceParam)
+	if namespace == "" {
+		if strings.Contains(rawEmulatorHostURL, ".") {
+			namespace = strings.Split(rawEmulatorHostURL, ".")[0]
+		}
```

Comment:
I observed similar behaviour in the Node sdk: https://github.com/firebase/firebase-admin-node/blob/master/src/database/database.ts#L325